### ISSUE TITLE
[MacCatalyst] OperatingSystem is currently not working 

### ISSUE THREAD

**Issue by rolfbjarne** [dotnet/runtime#108694]
_From @PureWeen on Tue, 08 Oct 2024 20:22:20 GMT_

### Description

It looks like on MacCatalyst the Build on the `Environment.OSVersion.Version` is set to "-1" so when you call 

`if (OperatingSystem.IsMacCatalystVersionAtLeast(18,0))`

it resolves to false. 
This works fine on iOS just not MacCatalyst

### Reproduction
[aoperatingsystem.zip](https://github.com/user-attachments/files/17298855/aoperatingsystem.zip)

Just run the app and I've set a label on the screen to output the value of `OSVersion.Version`. I've also included some if statements in the main page you can break point

#### Catalyst
![image](https://github.com/user-attachments/assets/e69cc7c1-d447-4a70-b444-682cfcf28d97)

#### iOS

![image](https://github.com/user-attachments/assets/f68e99f7-c867-4e17-b788-5e9b75093d08)

_Copied from original issue xamarin/xamarin-macios#21390_

**Comment by rolfbjarne** [comment]
_From @drasticactions on Wed, 09 Oct 2024 05:31:59 GMT_

TL;DR macOS reports `18.0`, SDK parses it as a `Version` which is where the `-1` comes from, Runtime check will always report it as below. 

https://github.com/drasticactions/MauiRepoRedux/tree/MacCatalystVersion

https://github.com/dotnet/runtime/blob/abde3f9498f09b5016a7a0d7d2e1b81ce5c1b614/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L227

<img width="759" alt="スクリーンショット 2024-10-09 14 19 32" src="https://github.com/user-attachments/assets/2a4f267f-b66c-4252-9022-750687810fb9">

https://github.com/dotnet/runtime/blob/abde3f9498f09b5016a7a0d7d2e1b81ce5c1b614/src/libraries/Common/src/Interop/OSX/System.Native/Interop.iOSSupportVersion.cs#L11

https://github.com/dotnet/runtime/blob/abde3f9498f09b5016a7a0d7d2e1b81ce5c1b614/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L328-L341

https://github.com/xamarin/xamarin-macios/blob/0db186c75896a516b409274b45136714230d60ab/src/ObjCRuntime/Runtime.cs#L2688-L2702

<img width="277" alt="image" src="https://github.com/user-attachments/assets/1ddc7f49-c2f4-4610-b8a1-649f06ec359b">

https://github.com/xamarin/xamarin-macios/blob/0db186c75896a516b409274b45136714230d60ab/tests/common/PlatformInfo.cs#L82

That value is coming from `SystemVersion.plist`, which is `18.0`, hence why it would get `-1`, since it's `Version.Parse`ed and it doesn't have the Build number that it may have had in the past.

https://github.com/dotnet/runtime/blob/abde3f9498f09b5016a7a0d7d2e1b81ce5c1b614/src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs#L344

It may be that this needed that the `-1` check needs to also be added to build for the same reason as revision, since it can't be assumed that build will be there. Or the SDK code above that fetches the version should always include the build even if the system doesn't provide it.

To work around it, checking against`18` `0` `-1` should work

**Comment by akoeplinger** [comment]
@am11 I think we should fix this in Environment.OSVersion.MacCatalyst.cs instead and default `patch` to 0. We already do that for e.g. `GetOperatingSystemVersion` in Interop.libobjc.cs

That said, I wonder how this ever worked since e.g. I also don't have a three-part version on my macOS 14.7 machine and from some quick googling it seems to always have been a two-part string?

**Comment by akoeplinger** [comment]
@vitek-karas we probably need to backport this.

**Comment by am11** [comment]
Yup, whichever solution makes sense. If it's a bit more general, may just cover more platforms. `Version.Parse` requires two parts (major and minor) at minimum, everything else is optional and result in -1.

**Comment by kotlarmilos** [comment]
The `IsOSVersionAtLeast​` fix will be backported to .NET 9 in a servicing release without the breaking change once the `release/9.0-staging` becomes available.

**Comment by akoeplinger** [comment]
This was backported in https://github.com/dotnet/runtime/pull/109332
